src/lib/client/GoodCityApiV2Client/GoodCityApiV2Client.test.ts

```diff
@@ -0,0 +1,79 @@
+import GoodCityApiV2Client from "./GoodCityApiV2Client";
+import sendPinResponse from "./__mocks__/sendPinResponse.js";
+import verifyResponse from "./__mocks__/verifyResponse.js";
+
+const mockFetch = jest.spyOn(window, "fetch");
+
+afterEach(() => mockFetch.mockClear());
+
+afterAll(() => mockFetch.mockRestore());
+
+describe("sendPin", () => {
+  it(`should call fetch with /auth/send_pin URL and the correct config`, () => {
+    const mobile = "+85262345678";
+
+    GoodCityApiV2Client.sendPin({ mobile });
+
+    expect(mockFetch).toHaveBeenCalledTimes(1);
+    expect(mockFetch).toHaveBeenCalledWith(
+      `${process.env.REACT_APP_API_V2_URL}/auth/send_pin`,
+      {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ mobile }),
+      }
+    );
+  });
+
+  describe("Successful API response", () => {
+    it("it should return the appropriate response", async () => {
+      mockFetch.mockResolvedValueOnce({
+        ok: true,
+        json: async () => Promise.resolve(sendPinResponse),
+      } as Response);
```

Comment:
Is it really ok to satisfy the type-checking in this way? What about other important fields like `status`? Should we look into a fetch mocking third party library to abstract away the tedium?